autoload/phpactor.vim

```diff
@@ -732,28 +797,7 @@ function! s:openFileInSelectedTarget(filePath, target, useOpenWindow, forceReloa
         endif
     endif
 
-    if a:target ==# 'focused_window'
-        call phpactor#_switchToBufferOrEdit(a:filePath)
-        if v:true == a:forceReload
-          exec 'e!'
-        endif
-        return
-    endif
-
-    if a:target ==# 'vsplit'
-        exec ':vsplit ' . a:filePath
-        return
-    endif
-
-    if a:target ==# 'hsplit'
-        exec ':split ' . a:filePath
-        return
-    endif
-
-    if a:target ==# 'new_tab'
-        exec ':tabnew ' . a:filePath
-        return
-    endif
+    execute printf('%s %s %s', a:mods, a:target, a:filePath)
```

Comment:
Can we validate the target here?

If I provide `tasd` as a target i get:

```
E492: Not an editor command:  tasd /home/daniel/www/phpactor/phpactor/vendor/psr/event-dispatcher/src/EventDispatcherInterface.php 
```

Can we instead throw an error listing possible targets:

```
invalid target "tasd` valid targets: "tabe", "vspl", // ... etc
```

Reply:
This is also important because I can pass "editor commands" e.g. `:PhpactorGotoDefinition bdelete`

Reply:
and if I execute:

```
:PhpactorGotoDefinition echo "asd"
```

I get:

```
E121: Undefined variable: home 
```

Reply:
I see your point but I don't know a way to validate against every possibilities offered by the editor.
So if we miss some a user will be limited (at least until we fix it), and we will have to maintain it all the way.
So I would argue that it's up to the user to check vim capabilities first, this is barely a wrapper allowing as mush flexibility as possible for the user.

Now if you feel that we should actually protect the user against himself, I guess we can reuse `s:windowTargets` to validate the targets.
But for the modifiers I don't thinks it's necessary, vim should fail by itself if provided with an invalid modifier to the command... I hope :)

In a way I think I didn't want to bother with that because vim does not, try `toto /tmp/file` and you got the same error.
I guess vim follow the unix philosophy saying "you now what you do" and I'm fine doing the same thing.

Let me know if you want phpactor to be more user friendly than "unix oriented" :D

Reply:
Well - this is an unhandled error (undefined variable) - we should handle the error. If we can suggest what values the user can use, then all the better.

Reply:
Done, check if it's OK with you.